File: backend/wifi-be/src/main/java/org/andreiz0r/wifibe/entity/WifiData.java
package org.andreiz0r.wifibe.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.util.UUID;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder(setterPrefix = "with")
@ToString
public class WifiData {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String ssid;

    @Column(nullable = false, unique = true)
    private String mac;

    @Column(nullable = false)
    private String clientIp;

    @Column(nullable = false)
    private String security;

    @Column(nullable = false)
    private int signalStrength;
}

File: mobile/lib/data/wifi_data.dart
class WifiData {
  final String ssid;
  final String mac;
  final String security;
  final int signalStrength;

  WifiData({
    required this.ssid,
    required this.mac,
    required this.security,
    required this.signalStrength,
  });

  factory WifiData.fromJson(Map<String, dynamic> json) {
    return WifiData(
      ssid: json['ssid'],
      mac: json['mac'],
      security: json['security'],
      signalStrength: json['signalStrength'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'ssid': ssid,
      'mac': mac,
      'security': security,
      'signalStrength': signalStrength,
    };
  }
}
